cli/src/main.rs

```diff
@@ -24,825 +27,49 @@ extern crate env_logger;
-				.options(|app| {
-					app.arg(
-						Arg::with_name("from")
-							.takes_value(true)
-							.required(true)
-							.value_name("SS58")
-							.help("Sender's on-chain AccountId in ss58check format"),
-					)
-					.arg(
-						Arg::with_name("to")
-							.takes_value(true)
-							.required(true)
-							.value_name("SS58")
-							.help("Recipient's incognito AccountId in ss58check format"),
-					)
-					.arg(
-						Arg::with_name("amount")
-							.takes_value(true)
-							.required(true)
-							.value_name("U128")
-							.help("Amount to be transferred"),
-					)
-					.arg(
-						Arg::with_name("shard")
-							.takes_value(true)
-							.required(true)
-							.value_name("STRING")
-							.help("Shard identifier"),
-					)
-				})
-				.runner(move |_args: &str, matches: &ArgMatches<'_>| {
-					let chain_api = get_chain_api(matches);
-					let amount = matches
-						.value_of("amount")
-						.unwrap()
-						.parse()
-						.expect("amount can't be converted to u128");
-
-					let shard_opt = match matches.value_of("shard") {
-						Some(s) => match s.from_base58() {
-							Ok(s) => ShardIdentifier::decode(&mut &s[..]),
-							_ => panic!("shard argument must be base58 encoded"),
-						},
-						_ => panic!(
-							"at least one of `mrenclave` or `shard` arguments must be supplied"
-						),
-					};
-					let shard = match shard_opt {
-						Ok(shard) => shard,
-						Err(e) => panic!("{}", e),
-					};
-
-					// get the sender
-					let arg_from = matches.value_of("from").unwrap();
-					let from = get_pair_from_str(arg_from);
-					let chain_api = chain_api.set_signer(sr25519_core::Pair::from(from));
-
-					// get the recipient
-					let arg_to = matches.value_of("to").unwrap();
-					let to = get_accountid_from_str(arg_to);
-					let (_to_encoded, to_encrypted) = match encode_encrypt(matches, to) {
-						Ok((encoded, encrypted)) => (encoded, encrypted),
-						Err(e) => panic!("{}", e),
-					};
-					// compose the extrinsic
-					let xt: UncheckedExtrinsicV4<([u8; 2], Vec<u8>, u128, H256)> = compose_extrinsic!(
-						chain_api,
-						TEEREX,
-						"shield_funds",
-						to_encrypted,
-						amount,
-						shard
-					);
-
-					let tx_hash =
-						chain_api.send_extrinsic(xt.hex_encode(), XtStatus::Finalized).unwrap();
-					println!("[+] TrustedOperation got finalized. Hash: {:?}\n", tx_hash);
-					Ok(())
-				}),
-		)
-		.add_cmd(ita_stf::cli::cmd(&perform_trusted_operation))
-		.no_cmd(|_args, _matches| {
-			println!("No subcommand matched");
-			Ok(())
-		})
-		.run();
-	if let Err(e) = res {
-		println!("{}", e)
-	}
+#[derive(Parser)]
+#[clap(name = "integritee-cli")]
+#[clap(version = VERSION)]
+#[clap(author = "Integritee AG <hello@integritee.network>")]
+#[clap(about = "interact with integritee-node and workers", long_about = None)]
+#[clap(after_help = "stf subcommands depend on the stf crate this has been built against")]
+pub struct Cli {
```

Comment:
Is `CLI` (Command Line Interface`) the correct name for this struct? Are these not rather the CLI arguments? When I see CLI, I expect something that I can interact with, an interface with functions to call.

Reply:
I took kind of the convention of clap for the "mother element". We can change it but in fact the struct is more than just the arguments. E.g. it also parses the input.

Reply:
Alright, if it's more than just the argument and clap has the same naming convention, we can leave it as is 👍